iina/AppDelegate.swift

```diff
@@ -573,13 +577,16 @@ class AppDelegate: NSObject, NSApplicationDelegate, SPUUpdaterDelegate {
         // All players have shutdown.
         Logger.log("All players have shutdown")
       }
-      guard !OnlineSubtitle.loggedIn else { return }
-      // All players have shutdown. No longer logged into an online subtitles provider.
+      // If still logged in to subtitle providers or still saving playback history then continue
+      // waiting.
+      guard !OnlineSubtitle.loggedIn, HistoryController.shared.tasksOutstanding == 0 else { return }
```

Comment:
I suggest using two separate guards.

Reply:
Fixed. I initially had it as two guards as I thought that was easier to read with separate comments. I decided to combine it into one guard right before posting as a lot of the code uses guards with multiple expressions. In this case with the separate guards it is quicker to read and understand.